### PR TITLE
"install_clion.sh" Script Improvements

### PR DESCRIPTION
**Pull request by garethellis0**
### Description
- Added the '-f' flag to `install_clion.sh` to overwrite existing symlinks
- Seperated CLion version into year, major, minor so it can be specified once and parts of it can be used throughout the script
- Moved symlinking to outside check for if CLion exists so that we always re-symlink (for example, if we change versions but have both versions installed already)
- Added the bazel version to generated directories, compressed files so that if the version is changed it's correctly re-downloaded
- Added a message indicating what folder to remove to force bazel plugin reinstall

### Testing Done
- tested installing different versions of CLion on my system, including ones I'd previously installed
- tested changing bazel plugin versions

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [x] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [x] **Remove all commented out code**
- [x] **Remove extra print statements**: for example, those just used for testing
- [x] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [x] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

**Comment by garethellis0** [comment]
Once this goes in, I'll make a post in #software noting the issues with the bazel plugin and indicating people can just re-run the `install_clion.sh` script on master to revert their CLion version and fix it.

**Comment by garethellis0** [comment]
Closing because the issue came back, looks like it takes some time? Will re-open if I have a fix.....

**Comment by garethellis0** [comment]
Re-opening with some improvements I made along the way to fixing the bazel plugin bug (that fix is here: https://github.com/UBC-Thunderbots/Software/pull/1019)

File: environment_setup/install_clion.sh [hunk 1]
```diff
@@ -15,49 +18,55 @@ then
...
+
 echo "================================================================"
 echo "Installing Bazel Plugin"
 echo "================================================================"
 
-clion_plugin_dir="${HOME}/.CLion2019.2/config/plugins"
-clion_major_version="2019.2"
-bazel_plugin_version="v2019.09.16"
+clion_plugin_dir="${HOME}/.CLion${clion_version_year}.${clion_version_major}/config/plugins"
+bazel_plugin_version="v2019.10.28"
```
Comment (MathewMacDougall) [inline environment_setup/install_clion.sh:46]: I'm assuming this newer plugin version is still compatible with CLion 2019.2.2?

**Review by MathewMacDougall — CHANGES_REQUESTED** [review]

File: environment_setup/install_clion.sh (hunk 1 above)
Comment (garethellis0) [reply environment_setup/install_clion.sh:46]: Yup, tested and using it. 2019.2.2 is still the newest version supported by the plugin

**Review by MathewMacDougall — APPROVED** [review]

**Review by matthewberends — APPROVED** [review]
LGTM

**Pull request merged**